mtp_noms_ops/apps/settings/views.py

```diff
@@ -152,3 +152,38 @@ def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         context['prisons'] = self.request.user_prisons
         return context
+
+
+class JobInformationView(FormView):
+    title = _('Help us improve this service')
+    template_name = 'settings/job-information.html'
+    success_url = reverse_lazy('security:dashboard')
+    form_class = JobInformationForm
+
+    def get_success_url(self):
+        if 'next' in self.request.GET:
+            return '{path}?{query}'.format(
+                path=self.success_url,
+                query=urlencode({'next': self.request.GET['next']})
+            )
+        return self.success_url
+
+    def form_valid(self, *args, **kwargs):
+        if has_provided_job_information(self.request.user):
+            return redirect(self.get_success_url())
+
+        if self.form_class(self.request.POST).is_valid():
+            session = get_api_session(self.request)
+            if self.request.POST['job_title'] == 'Other' and self.request.POST['other_title'] != '':
```

Comment:
you should be reading values from the validated form variable not the raw POST data

Reply:
also, i'd have this logic on the form class. e.g. the clean method could add another key to the cleaned_data dictionary

Reply:
Good plan, this makes it all simpler (at least no more complicated in the clean method and simpler in the view)